integration_tests/pages/page.ts

```diff
@@ -72,6 +72,22 @@ export default abstract class Page {
     })
   }
 
+  shouldContainErrorSummary(errors: Array<{ link: string; text: string }>): void {
+    cy.get('.govuk-error-summary').should('exist')
+    cy.get('.govuk-error-summary__list').within(() => {
+      cy.get('li').should('have.length', errors.length)
+      errors.forEach((error, index) => {
+        cy.get('li')
+          .eq(index)
+          .then(errorItem => {
+            const { actual, expected } = Helpers.parseHtml(errorItem, error.text)
+            expect(actual).to.equal(expected)
+            cy.get('a').should('have.attr', 'href', error.link)
+          })
+      })
```

Comment:
This is a bit different to how we've handled arrays in the past (e.g. in `shouldContainSummaryListRows` where we loop over the elements then index into the data array), but I quite like it. Didn't know about `.eq`

Reply:
Ah, I've actually used this now here: https://github.com/ministryofjustice/hmpps-accredited-programmes-ui/pull/184/files#diff-634ec1902ba20e2ee91b4806fe88435d5cae6cd75c5605c90b3604bfb58d70f8R173 😬

Felt a bit nicer than `nth-child(${horrible-string-interpolation})`. Don't think it matters too much though - `shouldHaveErrors` might replace the `shouldHaveErrorSummary` method